src/modules/adaptors/Curve/CurveAdaptor.sol

```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IWETH9 } from "src/interfaces/external/IWETH9.sol";
+import { CurvePool } from "src/interfaces/external/Curve/CurvePool.sol";
+import { CurveGauge } from "src/interfaces/external/Curve/CurveGauge.sol";
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { Cellar } from "src/base/Cellar.sol";
+import { CellarWithOracle } from "src/base/permutations/CellarWithOracle.sol";
+import { CurveHelper } from "src/modules/adaptors/Curve/CurveHelper.sol";
+
+// TODO Curve Gauges can TECHNICALLY have a non 18 decimal value :(
+// TODO should this validate positions are used?
+/**
+ * @title Curve Adaptor
+ * @notice Allows Cellars to interact with Curve LP positions.
+ * @author crispymangoes
+ */
+contract CurveAdaptor is BaseAdaptor, CurveHelper {
+    using SafeTransferLib for ERC20;
+    using Address for address;
+    using Strings for uint256;
+    using Math for uint256;
+
+    //==================== Adaptor Data Specification ====================
+    // adaptorData = abi.encode(address pool, address token, address gauge, bytes4 selector)
+    // Where:
+    // pool is the Curve Pool address
+    // token is the Curve LP token address(can be the same as pool)
+    // gauge is the Curve Gauge(can be zero address)
```

Comment:
So I allowed for a zero address because I am not confident in Curve always having a gauge for some LP, so this way we can work with LPs that dont have gauges